airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -355,6 +358,17 @@ private static void assertColumnsWithSameNameAreSame(String schemaName, String t
                 })
                 .collect(Collectors.toList())))
         .collect(Collectors.toList());
+    result.forEach(t -> {
+      try {
+        final List<String> primaryKeys = database.bufferedResultSetQuery(
+            conn -> conn.getMetaData().getPrimaryKeys(databaseOptional.orElse(null), t.getSchemaName(), t.getName()),
```

Comment:
do we need to do this query once per table? or if we leave table null, can we just run this query once for the whole database?

Reply:
I've tried to run the function without table names.

It seems like for MySQL this is not allowed and is throwing an exception about not having a table name but it does work for some other Databases... I made a new PR for this

See also:
> The method getTables takes a like-pattern for the tableNamePattern parameter, so "%" matches all table names.
> The method getPrimaryKeys and getExportedKeys do not take a pattern, so you will need to loop over the result of
getTables and execute those methods for each row of the getTables result set.

https://stackoverflow.com/a/40841486